api/base/filters.py

```diff
@@ -636,3 +637,24 @@ def get_serializer_method(self, field_name):
         serializer = self.get_serializer()
         serializer_method_name = 'get_' + field_name
         return getattr(serializer, serializer_method_name)
+
+
+class PreprintFilterMixin(DjangoFilterMixin):
+    """View mixin that uses DjangoFilterMixin, adding postprocessing for preprint querying
+
+       Subclasses must define `get_default_django_query()`.
+
+    """
+    def postprocess_query_param(self, key, field_name, operation):
+        if field_name == 'provider':
+            operation['source_field_name'] = 'provider___id'
+
+        if field_name == 'id':
+            operation['source_field_name'] = 'guids___id'
+
+        if field_name == 'subjects':
+            try:
+                Subject.objects.get(_id=operation['value'])
```

Comment:
Drive-by comment: It would be slightly more efficient to do 

```python
if Subject.objects.filter(_id=operation['value']).exists():
```

Since you don't need the actual subject objects, you save the overhead of deserializing the data to a `Subject`.